### ISSUE TITLE
Hexadecimal constants.

### ISSUE THREAD

**Issue by slembcke** [LukasBanana/XShaderCompiler#50]
I assume this is a known issue, but reporting it anyway. Hexidecimal numbers are parsed without error, but always end up as 0 in the output.

Ex:
```
const uint MANTISSA_MASK = 0x007FFFFFu;
const uint ONE = 0x3F800000u;
```

Both end up as "0u" in the GLSL output. Easily worked around with decimal constants, so it's not a big deal.

Very cool project by the way. I've really been enjoying playing around with it in my shader experiment project.

**Comment by LukasBanana** [comment]
Thanks for the hint, I'll take care of it as soon as I can.
Nice to see more people can make use of it :)

**Comment by LukasBanana** [comment]
Interestingly your example works for me:
```hlsl
const uint MANTISSA_MASK = 0x007FFFFFu;
const uint ONE = 0x3F800000u;
```
But the following results in the issue you described:
```hlsl
const int MANTISSA_MASK = 0x007FFFFFu;
const int ONE = 0x3F800000u;
```
Because here the compiler makes a conversion from the `0x007FFFFFu` literal to remove the `u` postfix character, which currently fails.

**Comment by slembcke** [comment]
Hmm. I'll have to double check when I get home, and maybe make a better replication sample. I didn't change the types when I copied pasted it, but maybe it was the code around it or something? Also, it was a global constant if that makes a difference.

**Comment by LukasBanana** [comment]
It's ok, the problem is the ~~`u` character which seems not to be supported~~ inside the [`std::stoll`](http://en.cppreference.com/w/cpp/string/basic_string/stol) function that is used to convert the literal string into a number.

**UPDATE:**
I used the `std::stoll` function wrong for hex numbers, the base must be 16 but I thought the function can determine this on its own.

**Comment by LukasBanana** [comment]
Fixed with [69ff23a](https://github.com/LukasBanana/XShaderCompiler/commit/69ff23a9a76c8f31ec438b52ad0e2d3098894a7d).
